### ISSUE TITLE
Remote end closed connection without response after successful headless log in with v2.8

### ISSUE THREAD

**Issue by apiology** [mintapi/mintapi#553]
**Describe the bug**

After the fixes in https://github.com/mintapi/mintapi/pull/552, running headless locally on macOS led to `urllib3.exceptions.ProtocolError: ('Connection aborted.', RemoteDisconnected('Remote end closed connection without response'))` after signin.

What version of MintAPI are you using? 

Local checkout of https://github.com/mintapi/mintapi/pull/552

What command[s] did you run / steps to reproduce?

```sh
mintapi --headless --beta --use-chromedriver-on-path username password
```

Stacktrace/error received:

```
Traceback (most recent call last):
  File "/Users/broz/.pyenv/versions/3.8.14/envs/mintapi-3.8.14/lib/python3.8/site-packages/urllib3/connectionpool.py", line 703, in urlopen
    httplib_response = self._make_request(
  File "/Users/broz/.pyenv/versions/3.8.14/envs/mintapi-3.8.14/lib/python3.8/site-packages/urllib3/connectionpool.py", line 449, in _make_request
    six.raise_from(e, None)
  File "<string>", line 3, in raise_from
  File "/Users/broz/.pyenv/versions/3.8.14/envs/mintapi-3.8.14/lib/python3.8/site-packages/urllib3/connectionpool.py", line 444, in _make_request
    httplib_response = conn.getresponse()
  File "/Users/broz/.pyenv/versions/3.8.14/Library/Frameworks/Python.framework/Versions/3.8/lib/python3.8/http/client.py", line 1348, in getresponse
    response.begin()
  File "/Users/broz/.pyenv/versions/3.8.14/Library/Frameworks/Python.framework/Versions/3.8/lib/python3.8/http/client.py", line 316, in begin
    version, status, reason = self._read_status()
  File "/Users/broz/.pyenv/versions/3.8.14/Library/Frameworks/Python.framework/Versions/3.8/lib/python3.8/http/client.py", line 285, in _read_status
    raise RemoteDisconnected("Remote end closed connection without"
http.client.RemoteDisconnected: Remote end closed connection without response

During handling of the above exception, another exception occurred:

Traceback (most recent call last):
  File "/Users/broz/.pyenv/versions/mintapi-3.8.14/bin/mintapi", line 33, in <module>
    sys.exit(load_entry_point('mintapi', 'console_scripts', 'mintapi')())
  File "/Users/broz/src/mintapi/mintapi/cli.py", line 518, in main
    data = mint.get_account_data(limit=options.limit)
  File "/Users/broz/src/mintapi/mintapi/api.py", line 265, in get_account_data
    return self.get_data(
  File "/Users/broz/src/mintapi/mintapi/api.py", line 245, in get_data
    data = self.__get_mint_endpoint(endpoint, **kwargs)
  File "/Users/broz/src/mintapi/mintapi/api.py", line 452, in __get_mint_endpoint
    response = self.get(
  File "/Users/broz/src/mintapi/mintapi/api.py", line 162, in get
    return self.driver.request("GET", url, **kwargs)
  File "/Users/broz/.pyenv/versions/3.8.14/envs/mintapi-3.8.14/lib/python3.8/site-packages/seleniumrequests/request.py", line 159, in request
    self.requests_session.headers = get_webdriver_request_headers(self, proxy_host=self.__proxy_host)
  File "/Users/broz/.pyenv/versions/3.8.14/envs/mintapi-3.8.14/lib/python3.8/site-packages/seleniumrequests/request.py", line 76, in get_webdriver_request_headers
    webdriver.switch_to.window(original_window_handle)
  File "/Users/broz/.pyenv/versions/3.8.14/envs/mintapi-3.8.14/lib/python3.8/site-packages/selenium/webdriver/remote/switch_to.py", line 134, in window
    self._w3c_window(window_name)
  File "/Users/broz/.pyenv/versions/3.8.14/envs/mintapi-3.8.14/lib/python3.8/site-packages/selenium/webdriver/remote/switch_to.py", line 142, in _w3c_window
    send_handle(window_name)
  File "/Users/broz/.pyenv/versions/3.8.14/envs/mintapi-3.8.14/lib/python3.8/site-packages/selenium/webdriver/remote/switch_to.py", line 138, in send_handle
    self._driver.execute(Command.SWITCH_TO_WINDOW, {'handle': h})
  File "/Users/broz/.pyenv/versions/3.8.14/envs/mintapi-3.8.14/lib/python3.8/site-packages/selenium/webdriver/remote/webdriver.py", line 427, in execute
    response = self.command_executor.execute(driver_command, params)
  File "/Users/broz/.pyenv/versions/3.8.14/envs/mintapi-3.8.14/lib/python3.8/site-packages/selenium/webdriver/remote/remote_connection.py", line 344, in execute
    return self._request(command_info[0], url, body=data)
  File "/Users/broz/.pyenv/versions/3.8.14/envs/mintapi-3.8.14/lib/python3.8/site-packages/selenium/webdriver/remote/remote_connection.py", line 366, in _request
    response = self._conn.request(method, url, body=body, headers=headers)
  File "/Users/broz/.pyenv/versions/3.8.14/envs/mintapi-3.8.14/lib/python3.8/site-packages/urllib3/request.py", line 78, in request
    return self.request_encode_body(
  File "/Users/broz/.pyenv/versions/3.8.14/envs/mintapi-3.8.14/lib/python3.8/site-packages/urllib3/request.py", line 170, in request_encode_body
    return self.urlopen(method, url, **extra_kw)
  File "/Users/broz/.pyenv/versions/3.8.14/envs/mintapi-3.8.14/lib/python3.8/site-packages/urllib3/poolmanager.py", line 376, in urlopen
    response = conn.urlopen(method, u.request_uri, **kw)
  File "/Users/broz/.pyenv/versions/3.8.14/envs/mintapi-3.8.14/lib/python3.8/site-packages/urllib3/connectionpool.py", line 787, in urlopen
    retries = retries.increment(
  File "/Users/broz/.pyenv/versions/3.8.14/envs/mintapi-3.8.14/lib/python3.8/site-packages/urllib3/util/retry.py", line 550, in increment
    raise six.reraise(type(error), error, _stacktrace)
  File "/Users/broz/.pyenv/versions/3.8.14/envs/mintapi-3.8.14/lib/python3.8/site-packages/urllib3/packages/six.py", line 769, in reraise
    raise value.with_traceback(tb)
  File "/Users/broz/.pyenv/versions/3.8.14/envs/mintapi-3.8.14/lib/python3.8/site-packages/urllib3/connectionpool.py", line 703, in urlopen
    httplib_response = self._make_request(
  File "/Users/broz/.pyenv/versions/3.8.14/envs/mintapi-3.8.14/lib/python3.8/site-packages/urllib3/connectionpool.py", line 449, in _make_request
    six.raise_from(e, None)
  File "<string>", line 3, in raise_from
  File "/Users/broz/.pyenv/versions/3.8.14/envs/mintapi-3.8.14/lib/python3.8/site-packages/urllib3/connectionpool.py", line 444, in _make_request
    httplib_response = conn.getresponse()
  File "/Users/broz/.pyenv/versions/3.8.14/Library/Frameworks/Python.framework/Versions/3.8/lib/python3.8/http/client.py", line 1348, in getresponse
    response.begin()
  File "/Users/broz/.pyenv/versions/3.8.14/Library/Frameworks/Python.framework/Versions/3.8/lib/python3.8/http/client.py", line 316, in begin
    version, status, reason = self._read_status()
  File "/Users/broz/.pyenv/versions/3.8.14/Library/Frameworks/Python.framework/Versions/3.8/lib/python3.8/http/client.py", line 285, in _read_status
    raise RemoteDisconnected("Remote end closed connection without"
urllib3.exceptions.ProtocolError: ('Connection aborted.', RemoteDisconnected('Remote end closed connection without response'))
```

**What did you expect to happen?**

Standard output

**What actually happened?**

Above exception

**Additional context**

As noted in https://github.com/mintapi/mintapi/pull/552, I appear to have some new HTML, at least on the signin side.

When not running headless, the error looks like https://github.com/mintapi/mintapi/issues/546 - so I would guess that this is related or a different way to reproduce the same error.

**Comment by ddollar** [comment]
I am seeing this same error, also headless

**Comment by mrooney** [comment]
Can you please try 2.9 from pypi and let me know if that resolves the issue? Thanks!

**Comment by ddollar** [comment]
I will try 2.9 but I have been seeing this issue on `main`

**Comment by ddollar** [comment]
@mrooney Same issue on 2.9

```
Traceback (most recent call last):
  File "/usr/local/lib/python3.9/dist-packages/urllib3/connectionpool.py", line 703, in urlopen
    httplib_response = self._make_request(
  File "/usr/local/lib/python3.9/dist-packages/urllib3/connectionpool.py", line 449, in _make_request
    six.raise_from(e, None)
  File "<string>", line 3, in raise_from
  File "/usr/local/lib/python3.9/dist-packages/urllib3/connectionpool.py", line 444, in _make_request
    httplib_response = conn.getresponse()
  File "/usr/lib/python3.9/http/client.py", line 1347, in getresponse
    response.begin()
  File "/usr/lib/python3.9/http/client.py", line 307, in begin
    version, status, reason = self._read_status()
  File "/usr/lib/python3.9/http/client.py", line 276, in _read_status
    raise RemoteDisconnected("Remote end closed connection without"
http.client.RemoteDisconnected: Remote end closed connection without response

During handling of the above exception, another exception occurred:

Traceback (most recent call last):
  File "/usr/local/bin/mintapi", line 8, in <module>
    sys.exit(main())
  File "/usr/local/lib/python3.9/dist-packages/mintapi/cli.py", line 518, in main
    data = mint.get_account_data(limit=options.limit)
  File "/usr/local/lib/python3.9/dist-packages/mintapi/api.py", line 262, in get_account_data
    return self.get_data(
  File "/usr/local/lib/python3.9/dist-packages/mintapi/api.py", line 242, in get_data
    data = self.__get_mint_endpoint(endpoint, **kwargs)
  File "/usr/local/lib/python3.9/dist-packages/mintapi/api.py", line 449, in __get_mint_endpoint
    response = self.get(
  File "/usr/local/lib/python3.9/dist-packages/mintapi/api.py", line 162, in get
    return self.driver.request("GET", url, **kwargs)
  File "/usr/local/lib/python3.9/dist-packages/seleniumrequests/request.py", line 159, in request
    self.requests_session.headers = get_webdriver_request_headers(self, proxy_host=self.__proxy_host)
  File "/usr/local/lib/python3.9/dist-packages/seleniumrequests/request.py", line 76, in get_webdriver_request_headers
    webdriver.switch_to.window(original_window_handle)
  File "/usr/local/lib/python3.9/dist-packages/selenium/webdriver/remote/switch_to.py", line 134, in window
    self._w3c_window(window_name)
  File "/usr/local/lib/python3.9/dist-packages/selenium/webdriver/remote/switch_to.py", line 142, in _w3c_window
    send_handle(window_name)
  File "/usr/local/lib/python3.9/dist-packages/selenium/webdriver/remote/switch_to.py", line 138, in send_handle
    self._driver.execute(Command.SWITCH_TO_WINDOW, {'handle': h})
  File "/usr/local/lib/python3.9/dist-packages/selenium/webdriver/remote/webdriver.py", line 427, in execute
    response = self.command_executor.execute(driver_command, params)
  File "/usr/local/lib/python3.9/dist-packages/selenium/webdriver/remote/remote_connection.py", line 344, in execute
    return self._request(command_info[0], url, body=data)
  File "/usr/local/lib/python3.9/dist-packages/selenium/webdriver/remote/remote_connection.py", line 366, in _request
    response = self._conn.request(method, url, body=body, headers=headers)
  File "/usr/local/lib/python3.9/dist-packages/urllib3/request.py", line 78, in request
    return self.request_encode_body(
  File "/usr/local/lib/python3.9/dist-packages/urllib3/request.py", line 170, in request_encode_body
    return self.urlopen(method, url, **extra_kw)
  File "/usr/local/lib/python3.9/dist-packages/urllib3/poolmanager.py", line 376, in urlopen
    response = conn.urlopen(method, u.request_uri, **kw)
  File "/usr/local/lib/python3.9/dist-packages/urllib3/connectionpool.py", line 787, in urlopen
    retries = retries.increment(
  File "/usr/local/lib/python3.9/dist-packages/urllib3/util/retry.py", line 550, in increment
    raise six.reraise(type(error), error, _stacktrace)
  File "/usr/local/lib/python3.9/dist-packages/urllib3/packages/six.py", line 769, in reraise
    raise value.with_traceback(tb)
  File "/usr/local/lib/python3.9/dist-packages/urllib3/connectionpool.py", line 703, in urlopen
    httplib_response = self._make_request(
  File "/usr/local/lib/python3.9/dist-packages/urllib3/connectionpool.py", line 449, in _make_request
    six.raise_from(e, None)
  File "<string>", line 3, in raise_from
  File "/usr/local/lib/python3.9/dist-packages/urllib3/connectionpool.py", line 444, in _make_request
    httplib_response = conn.getresponse()
  File "/usr/lib/python3.9/http/client.py", line 1347, in getresponse
    response.begin()
  File "/usr/lib/python3.9/http/client.py", line 307, in begin
    version, status, reason = self._read_status()
  File "/usr/lib/python3.9/http/client.py", line 276, in _read_status
    raise RemoteDisconnected("Remote end closed connection without"
urllib3.exceptions.ProtocolError: ('Connection aborted.', RemoteDisconnected('Remote end closed connection without response'))
```

**Comment by apiology** [comment]
@mrooney Confirmed that 2.9 does not resolve.  This occurs post-login, and not obviously tied to CSS selectors, so that is as expected.

**Comment by ddollar** [comment]
Have been doing some experimentation. The problem seems to only exist on chromedriver 106. If I revert to chromedriver 105 the problem goes away.

**Comment by pblesi** [comment]
I am also experiencing this issue on Ubuntu.

**Comment by felciano** [comment]
Confirming that 2.9 with a reversion to 105.0.5195.52 eliminates the issue when running in interactive mode. When running with `--headless` I'm oddly getting prompted for the "Mint e-mail:" despite passing it on the command line. Removing the `--headless` flag allows the command to proceed normally. 

@ddollar and @pblesi were you testing in interactive or headless mode?

**Comment by felciano** [comment]
This just started feeling on me again, November 1. The complaint now appears to be that Chrome is a different version down the Chromedriver. Do those two need to be in lockstep from aversion, standpoint? And if so, is there a clear way to specify a separate Chrome app instance to use or will it always use the system version?

**Comment by mpearon** [comment]
I'm having an issue similar.

|package|version|
|-----|-----|
|mintapi|2.9
|selenium-requests|2.0.0
|configargparse|1.5.3
|pandas|1.5.1
|keyring|23.5.0
|xmltodict|0.13.0
|oathtool|2.3.1
|requests|2.28.1
|selenium|4.6.0
|python-dateutil|2.8.1
|pytz|2022.1
|numpy|1.23.2
|trio|0.21.0
|certifi|2022.6.15
|urllib3[socks]|1.26.5
|trio-websocket|0.9.2
|tldextract|3.3.1
|charset-normalizer|2.1.1
|idna|3.3
|autocommand|2.2.1
|path|16.4.0
|filelock|3.8.0
|requests-file|1.5.1
|sortedcontainers|2.4.0
|attrs|22.1.0
|async-generator|1.10
|sniffio|1.3.0
|outcome|1.2.0
|wsproto|1.2.0
|PySocks|1.7.1
|six|1.16.0
|h11|0.13.0


```
mintapi --headless --no_wait_for_sync --accounts --transactions --include-investment --filename $mintTempDirectory my_username my_password
```

**Comment by apiology** [comment]
https://github.com/cryzed/Selenium-Requests/issues/53 has been fixed and released in selenium-requests 2.0.1.  I verified it resolves this issue for me.  Run `pip install --upgrade mintapi 'selenium-requests>=2.0.1'` to upgrade.

**Comment by mrooney** [comment]
I just released mintapi 2.10 to pypi, which updates selenium-requests to 2.0.1. Please try that (or update your selenium-requests to the same) and let me know if that resolves the issue. Thanks!
